core/mailer/mailer.go

```
@@ -0,0 +1,50 @@
+package mailer
+
+import (
+	"errors"
+	"net/smtp"
+	"os"
+	"regexp"
+
+	"github.com/joho/godotenv"
+)
+
+// Mailer contains context required to send a mail
+type Mailer struct {
+	toEmail, title, body string
+}
+
+// NewMailer returns a new Mailer using parameter target email, title, and body
+func NewMailer(email, title, body string) (*Mailer, error) {
+	emailFormat := regexp.MustCompile(`^[_a-z0-9-]+(\.[_a-z0-9-]+)*@[a-z0-9-]+(\.[a-z0-9-]+)*(\.[a-z]{2,4})$`)
+	if !emailFormat.MatchString(email) {
+		return nil, errors.New("mail: misformatted source email address")
+	}
+
+	return &Mailer{email, title, body}, nil
+}
+
+// Send attempts to send a email to the target email using struct information
+func (m *Mailer) Send() error {
+	if err := godotenv.Load(); err != nil {
+		return err
```

I would actually load dotenv only once at program init, and either a. ignore errors or b. log errors, then ignore them. This will allow us to provide values directly from env if desired